### PR TITLE
UI code cleanup

### PR DESCRIPTION
**Pull request by mvolkmann**
This adds the use of Prettier for code formatting and fixes all the issues that ESLint identified.

### CODE REVIEW

File: client/.eslintrc.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "eol-last": "off",
     "eqeqeq": "error",
     "flowtype/boolean-style": ["error", "boolean"],
-    "flowtype/define-flow-type": ["error", {"no-undef": "error"}],
```
Comment (mvolkmann) [inline client/.eslintrc.json:68]: The valid values for this ESLint rule changed.

File: client/.eslintrc.json [hunk 2]
```diff
@@ -94,37 +94,6 @@
     "id-blacklist": ["error"],
     "id-length": "off",
     "id-match": "off",
-    /*
```
Comment (mvolkmann) [inline client/.eslintrc.json:97]: This block of rules wasn't being used.

File: client/.eslintrc.json [hunk 3]
```diff
@@ -280,13 +248,13 @@
...
     "no-with": "error",
     "object-curly-newline": "off",
-    "object-curly-spacing": ["error", "always"],
+    "object-curly-spacing": ["error", "never"],
     "object-property-newline": "off",
     "object-shorthand": ["error", "always"],
     "one-var": "off",
     "one-var-declaration-per-line": "off",
     "operator-assignment": ["error", "always"],
-    "operator-linebreak": ["error", "after"],
```
Comment (mvolkmann) [inline client/.eslintrc.json:289]: This conflicted with Prettier.

File: client/.eslintrc.json [hunk 4]
```diff
@@ -280,13 +248,13 @@
     "no-whitespace-before-property": "error",
     "no-with": "error",
     "object-curly-newline": "off",
-    "object-curly-spacing": ["error", "always"],
```
Comment (mvolkmann) [inline client/.eslintrc.json:283]: This conflicted with Prettier.

File: client/.eslintrc.json [hunk 5]
```diff
@@ -295,12 +263,7 @@
     "prefer-rest-params": "error",
     "prefer-spread": "error",
     "prefer-template": "off",
-    "prettier/prettier": ["error", {
```
Comment (mvolkmann) [inline client/.eslintrc.json:298]: This is now configured in `.prettierrc`.

File: client/.eslintrc.json [hunk 6]
```diff
@@ -326,7 +289,6 @@
     "react/jsx-no-undef": "error",
     "react/jsx-pascal-case": "error",
     "react/jsx-sort-props": "off",
-    "react/jsx-space-before-closing": [1, "always"],
```
Comment (mvolkmann) [inline client/.eslintrc.json:329]: This rule was deprecated.

File: client/.flowconfig [hunk 7]
```diff
@@ -0,0 +1,9 @@
+[ignore]
```
Comment (mvolkmann) [inline client/.flowconfig:1]: This file is required to use Flow. We're not using it yet, but this prepares us.

File: client/package.json [hunk 8]
```diff
@@ -6,30 +6,32 @@
     "bootstrap": "^4.0.0-alpha.6",
     "chai": "^4.1.0",
     "classnames": "^2.2.5",
-    "enzyme": "^3.0.0",
```
Comment (mvolkmann) [inline client/package.json:9]: Lots of package versions were updated.

File: client/package.json [hunk 9]
```diff
@@ -41,13 +43,15 @@
   },
   "dependencies": {},
   "scripts": {
+    "build": "npm run build-css && react-scripts build",
     "build-css": "node-sass-chokidar src/ -o src/",
-    "watch-css": "npm run build-css && node-sass-chokidar src/ -o src/ --watch --recursive",
```
Comment (mvolkmann) [inline client/package.json:45]: Alphabetized these.

File: client/src/App.js [hunk 10]
```diff
@@ -1,84 +1,108 @@
-import React, { Component } from "react";
```
Comment (mvolkmann) [inline client/src/App.js:1]: Lots of Prettier/ESLint updates from here on.

File: client/src/AppNav.js [hunk 11]
```diff
@@ -1,10 +1,20 @@
...
-function AppNav({ serverInfo }) {
+function getMenuItems(plugins) {
+  if (!plugins) return null;
+
+  return plugins.map(plugin => (
+    <MenuItem eventKey="4.1" key={plugin.name}>
+      {plugin.name} - {plugin.version}
+    </MenuItem>
+  ));
+}
```
Comment (mkimberlin) [inline client/src/AppNav.js:15]: Just for the sake of comparison...I would have written this:
```
const getMenuItems = plugins => plugins && plugins.map(plugin => (
  <MenuItem eventKey="4.1" key={plugin.name}>
    {plugin.name} - {plugin.version}
  </MenuItem>
));
```

**Review by mkimberlin — APPROVED** [review]

**Pull request merged**